### ISSUE TITLE
Check constructor arguments when calling Activator.CreateInstance

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/ReflectionAnalyzers#207]
```cs
return (EqualByComparer)Activator.CreateInstance(
    typeof(ReadOnlyListEqualByComparer<>).MakeGenericType(itemType),
    settings.GetEqualByComparer(itemType),
    ↓typeof(int));
```

Should warn there as:

```cs
    internal class ReadOnlyListEqualByComparer<T> : EqualByComparer
    {
        private readonly EqualByComparer<T> comparer;

        public ReadOnlyListEqualByComparer(EqualByComparer<T> comparer)
        {
            this.comparer = comparer;
        }
```
